lib/ui/window.dart

```diff
@@ -294,23 +357,31 @@ class Locale {
       return false;
     final Locale typedOther = other;
     return languageCode == typedOther.languageCode
-        && countryCode == typedOther.countryCode;
+        && scriptCode == typedOther.scriptCode
+        && countryCode == typedOther.countryCode
+        && _variants == typedOther._variants;
   }
 
   @override
   int get hashCode {
     int result = 373;
     result = 37 * result + languageCode.hashCode;
-    if (_countryCode != null)
-      result = 37 * result + countryCode.hashCode;
+    result = 37 * result + scriptCode.hashCode;
+    result = 37 * result + countryCode.hashCode;
+    result = 37 * result + _variants.hashCode;
     return result;
   }
 
   @override
   String toString() {
-    if (_countryCode == null)
-      return languageCode;
-    return '${languageCode}_$countryCode';
+    final StringBuffer out = StringBuffer(languageCode);
+    if (scriptCode != null)
+      out.write('-$scriptCode');
+    if (_countryCode != null)
+      out.write('-$countryCode');
```

Comment:
Discussion point: do we dare change the underscore to hyphen? Does this belong in a separate change?

Key data point: https://flutter.io/tutorials/internationalization/ suggests passing toString() to Intl.canonicalizedLocale. canonicalizedLocale handles hyphens and underscores, both are fine:
https://docs.flutter.io/flutter/intl/Intl/canonicalizedLocale.html

This change might still break other code that's gotten naive about Locale strings though.

Reply:
ignore that tutorial, we'll fix it once Intl is revamped. Code shouldn't be using toString at all. Ideally the toString of the Locale would be non-machine-readable, it'd just be a debug string.

Reply:
(but we shouldn't change toString output for existing inputs, especially if doing so would encourage more use)

Reply:
Switching back to underscores for now, postponing this discussion until later / out of scope for this pull request.